tests/test_interrupted_sync.py

```diff
@@ -0,0 +1,185 @@
+from tap_tester import runner, connections, menagerie
+from base import TiktokBase
+from dateutil import parser as parser
+
+
+class TiktokAdsInterruptedSyncTest(TiktokBase):
+    """
+    Test to verify that if a sync is interrupted, then the next sync will continue
+    from the bookmarks and currently syncing stream.
+    """
+
+    def name(self):
+        """ Returns the test name """
+        return "tap_tester_tiktok_interrupted_sync_test"
+
+    def test_run(self):
+        """
+        Scenario: A sync job is interrupted. The state is saved with `currently_syncing`.
+                  The next sync job kicks off and the tap picks back up on that
+                  `currently_syncing` stream.
+
+        Expected State Structure:
+            {
+                "currently_syncing": "stream_name",
+                "bookmarks": {
+                    "stream_1": {"bookmark_field": "2010-10-10T10:10:10.100000"},
+                    "stream_2": {"bookmark_field": "2010-10-10T10:10:10.100000"}
+                }
+            }
+
+        Test Cases:
+        - Verify an interrupted sync can resume based on the `currently_syncing` and
+            stream level bookmark value.
+        - Verify only records with replication-key values greater than or equal to the
+            stream level bookmark are replicated on the resuming sync for the interrupted stream.
+        - Verify the yet-to-be-synced streams are replicated following the interrupted stream
+            in the resuming sync.
+        """
+
+        self.start_date = self.get_properties()["start_date"]
+        start_date_datetime = self.dt_to_ts(self.start_date)
+
+        conn_id = connections.ensure_connection(self)
+        expected_streams = self.expected_streams() - {"advertisers"}
+
+        # Run check mode
+        found_catalogs = self.run_and_verify_check_mode(conn_id)
+
+        # Select only the expected streams tables
+        catalog_entries = [ce for ce in found_catalogs if ce["tap_stream_id"] in expected_streams]
+
+        # Catalog selection
+        self.select_found_catalogs(conn_id, catalog_entries)
+
+        # Run a sync job
+        self.run_and_verify_sync(conn_id)
+        first_full_sync = runner.get_records_from_target_output()
+
+        first_full_sync_state = menagerie.get_state(conn_id)
+
+        # State to run 2nd sync
+        interrupted_sync_state = {
+            "currently_syncing": "ad_insights",
+            "bookmarks": {
+                "campaigns": {
+                    "7086361182503780353": "2022-04-21T09:35:11.000000Z"
+                },
+                "adgroups": {
+                    "7086361182503780353": "2022-04-21T09:36:19.000000Z"
+                },
+                "ads": {
+                    "7086361182503780353": "2022-04-21T09:06:35.000000Z"
+                },
+                "ad_insights": {
+                    "7086361182503780353": "2021-01-25T00:00:00.000000Z"
+                }
+            }
+        }
```

Comment:
if possible can this state be created dynamically? So that there won't be any changes in test cases in near future for any new strem introduced.